### PR TITLE
Implemetation of WriteBack for applied styles into stylsheet

### PR DESCRIPTION
**Pull request by lschirmbrand**

### CODE REVIEW

File: src/elements/services/stylesheetService/StylesheetService.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        if (index > -1) highestSpecificityRule.block.children.splice(index, 1, (newRule.children[0] as csstree.RulePlain).block.children[0]);
+        else highestSpecificityRule.block.children.push((newRule.children[0] as csstree.RulePlain).block.children[0]);
+
+        if (!this.ruleset) this.ruleset = this.parseStylesheetToRuleset(this.stylesheet);
+
+        this.stylesheetChanged.emit({ stylesheet: csstree.generate(csstree.fromPlainObject(this.ruleset)) });
+        return true;
+    }
+
+    /*
```
Comment (lschirmbrand) [inline src/elements/services/stylesheetService/StylesheetService.ts:40]: We need to discuss further action, regarding my code comment

**Comment by jogibear9988** [comment]
https://github.com/node-projects/web-component-designer/issues/213

File: src/elements/services/stylesheetService/StylesheetService.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+export class StylesheetService implements IStylesheetService {
+
+    stylesheet: string;
+    stylesheetChanged: TypedEvent<{ stylesheet: string; }> = new TypedEvent<{ stylesheet: string; }>();
+
+    ruleset: csstree.StyleSheetPlain;
+
+    constructor(stylesheet: string) {
```
Comment (jogibear9988) [inline src/elements/services/stylesheetService/StylesheetService.ts:20]: string[] for more stylesheets,
maybe also an object like {name:string, stylesheet:string}[]

File: src/elements/services/stylesheetService/IStylesheetService.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { TypedEvent } from "@node-projects/base-custom-webcomponent";
+import { IDesignItem } from "../../item/IDesignItem.js";
+import { IProperty } from "../propertiesService/IProperty.js";
+import * as csstree from 'css-tree';
+
+export interface IStylesheetService {
+    getDefiningRule(designItems: IDesignItem[], property: IProperty): csstree.RulePlain;
```
Comment (jogibear9988) [inline src/elements/services/stylesheetService/IStylesheetService.ts:7]: we should not return types from external libs, cause we could switch to other implenentations
they should be a internal impl. detail

File: src/elements/services/stylesheetService/IStylesheetService.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+    updateDefiningRule(designItems: IDesignItem[], prop: IProperty, value: string): boolean;
+    stylesheetChanged: TypedEvent<{ stylesheet: string }>;
+}
```
Comment (jogibear9988) [inline src/elements/services/stylesheetService/IStylesheetService.ts:10]: we also need a api:

getAllRules ... (or smth similar)

File: src/elements/services/propertiesService/services/AbstractPropertiesService.ts [hunk 5]
```diff
@@ -27,14 +27,17 @@ export abstract class AbstractPropertiesService implements IPropertiesService {
     const cg = designItems[0].openGroup("properties changed");
     for (let d of designItems) {
       if (property.propertyType == PropertyType.cssValue) {
-        if (d.getStyle(property.name) != value) {
-          d.setStyle(property.name, value);
-        } else {
-          this.clearValue(designItems, property)
+        const declaration = designItems[0].serviceContainer.stylesheetService?.updateDefiningRule(designItems, property, value);
```
Comment (jogibear9988) [inline src/elements/services/propertiesService/services/AbstractPropertiesService.ts:30]: we need a api for this in designitem

Comment (jogibear9988) [reply src/elements/services/propertiesService/services/AbstractPropertiesService.ts:30]: maybe like: setStyleInStylesheetOrLocal

**Pull request merged**